src/cobra/io/yaml.py

```diff
@@ -1,89 +1,111 @@
-# -*- coding: utf-8 -*-
-
-from __future__ import absolute_import
+"""Provide functions for I/O in YAML format."""
 
 import io
+from typing import TYPE_CHECKING, Any, Dict, Optional
 
 from ruamel.yaml.compat import StringIO
 from ruamel.yaml.main import YAML
 
-from cobra.io.dict import model_from_dict, model_to_dict
+from .dict import model_from_dict, model_to_dict
+
+
+if TYPE_CHECKING:
+    from cobra import Model
 
 
 YAML_SPEC = "1.2"
 
 
-class MyYAML(YAML):
-    def dump(self, data, stream=None, **kwargs):
-        inefficient = False
+class CobraYAML(YAML):
+    """Define custom subclass for YAML I/O."""
+
+    def dump(self, data: Dict, stream: Optional[Any] = None, **kwargs: Any) -> str:
```

Comment:
I think the type of `stream` can be https://docs.python.org/3/library/io.html#io.TextIOBase rather than `Any`.

Reply:
What about we use something more specific: https://docs.python.org/3/library/io.html#io.StringIO ?

Reply:
Okay sounds fair. I'll change it.